File: api/src/main/java/br/inventory/control/api/model/Product.java
package br.inventory.control.api.model;

import jakarta.persistence.*;
import lombok.Data;
import java.math.BigDecimal;

@Entity
@Table(name = "products")
@Data
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    private BigDecimal unitPrice;
    private String unitOfMeasure;
    private int quantityInStock;
    private int minStockQuantity;
    private int maxStockQuantity;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id", nullable = false)
    private Category category;
}

File: admin-front-end/src/types/product.ts
export interface Product {
  id: string;
  name: string;
  unitPrice: number;
  unitOfMeasure: string;
  quantityInStock: number;
  minStockQuantity: number;
  maxStockQuantity: number;
  categoryId: number;
  category?: {
    id: string;
    name: string;
  };
}

export interface CreateProductData {
  name: string;
  unitPrice: number;
  unitOfMeasure: string;
  quantityInStock: number;
  minStockQuantity: number;
  maxStockQuantity: number;
  categoryId: number;
}
